src/app/components/button/button.ts

```diff
@@ -238,13 +240,14 @@ export class ButtonDirective implements AfterViewInit, OnDestroy {
             (click)="onClick.emit($event)"
             (focus)="onFocus.emit($event)"
             (blur)="onBlur.emit($event)"
+            [ngStyle]="isIconPosRight ? { display: 'flex', flexDirection: 'row-reverse' } : null"
```

Comment:
instead of `isIconPosRight ` and all it's logic, it might be enough to put: 
```
p-button[iconPos='right'] spinnericon {
    order: 1;
}
```
in the button css

Reply:
> instead of `isIconPosRight ` and all it's logic, it might be enough to put:
> 
> ```
> p-button[iconPos='right'] spinnericon {
>     order: 1;
> }
> ```
> 
> in the button css

If I do that, I must edit the primeng-sass-theme project because is who generate all the styles and create another PR for that

Reply:
You are correct. 
I guess it should be up to the Prime team to decide if this should be resolved at theme level via css only or at component level here.

Reply:
> You are correct. I guess it should be up to the Prime team to decide if this should be resolved at theme level via css only or at component level here.

Yes, maybe is a good idea. I will mention one mod to ask how to proceed :)

Reply:
Hi, 

It's okay to make that by CSS

Reply:
Changed! :)